File: CS328_Distributed_and_Cloud_Computing/DCC_notes/files/lab5/rpc_grpc_hello_world/hw_client.py
import grpc
from assistant_pb2 import GreetRequest, MultRequest
from assistant_pb2_grpc import AssistantServiceStub

def run():
  with grpc.insecure_channel('localhost:8082') as channel:
    stub = AssistantServiceStub(channel)
    # Greeting
    print('> Greet: Hello Assistant?')
    res = stub.GreetWithInfo(GreetRequest(user_name='Peter', institution='SUSTech'))
    print(f'> Client received:\n{res}')
    # Multiplication
    print('> Mult: Requesting a multiplication task')
    res = stub.Multiply(MultRequest(xin=3.5, yin=5))
    print(f'> Client received:\n{res}')
  
if __name__ == '__main__':
  run()

File: CS328_Distributed_and_Cloud_Computing/DCC_notes/files/lab6/rpc_grpc_cross_language/client/hw_client.go
package main

import (
	"context"
	"dncc/grpc/pb"
	"fmt"
	"log"
	"os"

	"google.golang.org/grpc"
	"google.golang.org/grpc/credentials/insecure"
)

func main() {
	// Create a gRPC channel with no SSL/TLS configured.
	conn, err := grpc.NewClient("127.0.0.1:8082", grpc.WithTransportCredentials(insecure.NewCredentials()))
	if err != nil {
		log.Fatalf("did not connect: %v", err)
	}
	// Specify to close the gRPC channel later (when the function finishes)
	defer conn.Close()
	// Create a client stub
	stub := pb.NewAssistantServiceClient(conn)
	// Greeting
	fmt.Println("> Greet: Hello Assistant?")
	greetRes, err := stub.GreetWithInfo(context.Background(), &pb.GreetRequest{
		UserName:    "Peter",
		Institution: "SUSTech",
	})
	if err != nil {
		fmt.Printf("> Failed to greet: %v", err)
		os.Exit(0)
	}
	fmt.Printf("Client received:\n%v\n", greetRes)
	// Multiplication
	fmt.Println("> Mult: Requesting a multiplication task")
	multRes, err := stub.Multiply(context.Background(), &pb.MultRequest{
		Xin: 3.5,
		Yin: 5,
	})
	fmt.Printf("> Client received:\n%v\n", multRes)
}
